generated/nixnet/nixnet.proto

```diff
@@ -844,6 +809,84 @@ enum WriteState {
   WRITE_STATE_STATE_ETHERNET_WAKE = 1245317;
 }
 
+message FlexRayStats {
+    uint32 num_syntax_error_ch_a = 1;
+    uint32 num_syntax_error_ch_b = 2;
+    uint32 num_content_error_ch_a = 3;
+    uint32 num_content_error_ch_b = 4;
+    uint32 num_slot_boundary_violation_ch_a = 5;
+    uint32 num_slot_boundary_violation_ch_b = 6;
+}
+
+message J1939CommState{
+    uint32 pgn = 1;
+    uint32 source_address = 2;
+    uint32 destination_address = 3;
+    uint32 transmit_error = 4;
+    uint32 receive_error = 5;
+}
+
+message CanComm{
+    uint32 comm_state = 1;
+    uint32 transceiver_error = 2;
+    uint32 sleep = 3;
+    uint32 last_error = 4;
+    uint32 transmit_error_counter = 5;
+    uint32 receive_error_counter = 6;
+}
+
+message FlexRayComm{
+    uint32 poc_state = 1;
+    uint32 clock_correction_failed = 2;
+    uint32 passive_to_active_count = 3;
+    uint32 channel_a_sleep = 4;
+    uint32 channel_b_sleep = 5;
+}
+
+message LinComm{
+    uint32 sleep = 1;
+    uint32 comm_state = 2;
+    uint32 last_error = 3;
+    uint32 last_error_received = 4;
+    uint32 last_error_expected = 5;
+    uint32 last_error_id = 6;
+    uint32 transceiver_ready = 7;
+    uint32 schedule_index = 8;
+}
+
+message ReadStateValue {
+    oneof value{
+        uint64 time_current = 1;
+        uint64 time_communicating = 2;
+        uint64 time_start = 3;
+        CanComm can_comm = 4;
+        FlexRayComm flex_ray_comm = 5;
+        FlexRayStats flex_ray_stat = 6;
+        LinComm lin_comm = 7;
+        uint32 session_info = 8;
+        J1939CommState comm_stat = 9;
+        TimeLocalNetwork time_current2 = 10;
+        TimeLocalNetwork time_communicating2 = 11;
+        TimeLocalNetwork time_start2 = 12;
+        bytes raw_value = 13;
```

Comment:
This should be `state_value_raw` and should be directly on response message so that users can either access `read_state_response->state_value()-><one-of-fields>` or `read_state_response->state_value_raw()`

Reply:
Do you also need a `_raw` option in `WriteStateValue`? In that case it would be part of the `oneof`.

Reply:
- I have left the `state_value_raw` in the ReadStateValue message but I moved it out of the oneof.
- The reason for having raw value for ReadState is because we are manipulating the original value returned by the API using macros. So we have the raw value in a separate field in case the user requires the whole value.
- For WriteState, StateValue is an input parameter and we are not manipulating the value in any way. So as of now, we dont have any use case for having a raw value.

Reply:
- I have left the `state_value_raw` in the ReadStateValue message but I moved it out of the oneof.
- The reason for having raw value for ReadState is because we are manipulating the original value returned by the API using macros. So we have the raw value in a separate field in case the user requires the whole value.
- For WriteState, StateValue is an input parameter and we are not manipulating the value in any way. So as of now, we dont have any use case for having a raw value.